homeassistant/components/risco/binary_sensor.py

```diff
@@ -82,15 +88,71 @@ def device_class(self):
         """Return the class of this sensor, from BinarySensorDeviceClass."""
         return BinarySensorDeviceClass.MOTION
 
-    async def _bypass(self, bypass):
-        alarm = await self._risco.bypass_zone(self._zone_id, bypass)
-        self._zone = alarm.zones[self._zone_id]
-        self.async_write_ha_state()
-
     async def async_bypass_zone(self):
         """Bypass this zone."""
         await self._bypass(True)
 
     async def async_unbypass_zone(self):
         """Unbypass this zone."""
         await self._bypass(False)
+
+    async def _bypass(self, bypass):
+        raise NotImplementedError
+
+
+class RiscoCloudBinarySensor(RiscoEntity, RiscoBinarySensor):
+    """Representation of a Risco cloud zone as a binary sensor."""
+
+    def __init__(self, coordinator, zone_id, zone):
+        """Init the zone."""
+        RiscoEntity.__init__(self, coordinator)
+        RiscoBinarySensor.__init__(self, zone_id, zone)
```

Comment:
It's better to pass on not used parameters to the parent via `super()` than calling different parents specifically.

Reply:
Not sure I understand - is there an example?

Reply:
I'd do it like this:

```py
class RiscoEntity(CoordinatorEntity[RiscoDataUpdateCoordinator]):
    """Risco entity base class."""

    ...

...


class RiscoBinarySensor(BinarySensorEntity):
    """Representation of a Risco zone as a binary sensor."""

    def __init__(self, *, zone_id: int, zone: Zone, **kwargs: Any) -> None:
        """Init the zone."""
        super().__init__(**kwargs)
        self._zone_id = zone_id
        self._zone = zone

    ...


class RiscoCloudBinarySensor(RiscoBinarySensor, RiscoEntity):
    """Representation of a Risco cloud zone as a binary sensor."""

    def __init__(
        self, coordinator: RiscoDataUpdateCoordinator, zone_id: int, zone: Zone
    ) -> None:
        """Init the zone."""
        super().__init__(zone_id=zone_id, zone=zone, coordinator=coordinator)

    ...


class RiscoLocalBinarySensor(RiscoBinarySensor):
    """Representation of a Risco local zone as a binary sensor."""

    def __init__(
        self,
        system_id: str,
        zone_id: int,
        zone: Zone,
        zone_updates: dict[int, Callable[[], Any]],
    ) -> None:
        """Init the zone."""
        super().__init__(zone_id=zone_id, zone=zone)
        self._system_id = system_id
        self._zone_updates = zone_updates

    ...
```

Reply:
thank you!